pkg/controller/nodelifecycle/scheduler/taint_manager.go

```diff
@@ -115,6 +119,27 @@ func deletePodHandler(c clientset.Interface, emitEventFunc func(types.Namespaced
 	}
 }
 
+func addConditionAndDeletePod(ctx context.Context, c clientset.Interface, name, ns string) (err error) {
+	if feature.DefaultFeatureGate.Enabled(features.JobPodFailurePolicy) {
+		pod, err := c.CoreV1().Pods(ns).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		newStatus := pod.Status.DeepCopy()
+		if apipod.UpdatePodCondition(newStatus, &v1.PodCondition{
+			Type:    "TargettedForDisruption",
+			Status:  v1.ConditionTrue,
+			Reason:  "DeletedByTaintManager",
```

Comment:
Maybe just `ByTaintManager` or `DeletionByTaintManager`, because the actual deletion is coming later.

The same can be said for the message.

Reply:
I prefer "DeletionByTaintManager" over "ByTaintManager", as it is a proper reason description without the context. It also feels more consistent naming with the reasons introduced by Kubelet (which include the component name and a short reason description, like KubeletHasInsufficientMemory): https://github.com/kubernetes/kubernetes/blob/c633f7124d9125067d89bd1e92cdff882144ced6/pkg/kubelet/nodestatus/setters.go#L594